normandy/recipes/tests/test_commands.py

```diff
@@ -257,3 +258,42 @@ def test_it_does_not_resign_up_to_date_actions(self, settings, mocked_autograph)
         call_command("update_action_signatures")
         a.refresh_from_db()
         assert a.signature.signature == "original signature"
+
+
+@pytest.mark.django_db
+class TestUpdateAddonUrls(object):
+    def test_it_works(self):
+        action = ActionFactory(name="opt-out-study")
+        recipe = RecipeFactory(
+            action=action,
+            arguments={"addonUrl": "https://before.example.com/extensions/addon.xpi"},
+        )
+        call_command("update_addon_urls", "after.example.com")
+        recipe = Recipe.objects.get(id=recipe.id)
+        assert recipe.arguments["addonUrl"] == "https://after.example.com/extensions/addon.xpi"
+
+    def test_signatures_are_updated(self, mocked_autograph):
+        action = ActionFactory(name="opt-out-study")
+        recipe = RecipeFactory(
+            action=action,
+            arguments={"addonUrl": "https://before.example.com/extensions/addon.xpi"},
+            approver=UserFactory(),
+            enabler=UserFactory(),
+            signed=True,
+        )
+        # preconditions
+        assert recipe.signature is not None
+        assert recipe.signature.signature == hashlib.sha256(recipe.canonical_json()).hexdigest()
+        signature_before = recipe.signature.signature
+
+        call_command("update_addon_urls", "after.example.com")
+        recipe.refresh_from_db()
+
+        assert recipe.signature is not None
+        assert recipe.signature != signature_before
+        assert recipe.signature.signature == hashlib.sha256(recipe.canonical_json()).hexdigest()
+
+    def test_new_hostname_is_required(self):
```

Comment:
This test is not needed. It's just testing Django. Not your code.

Reply:
I actually wrote this test before I wrote the command. It is mostly testing that I correctly remembered how to set `required=True` on the argparser. I'd prefer to keep it.